core/src/components/searchbar/searchbar.tsx

```diff
@@ -346,15 +362,30 @@ export class Searchbar implements ComponentInterface {
     return this.value || '';
   }
 
+  private hasValue(): boolean {
+    return this.getValue() !== '';
+  }
+
+  private emitStyle() {
+    this.ionStyle.emit({
+      'interactive': true,
+      'input': true,
+      'searchbar-has-value': this.hasValue(),
+      'searchbar-has-focus':  this.focused,
+      'interactive-disabled': this.disabled
+    });
+  }
+
   hostData() {
     const animated = this.animated && this.config.getBoolean('animated', true);
 
     return {
+      'searchbar-disabled': this.disabled ? 'true' : null,
       class: {
         ...createColorClasses(this.color),
         'searchbar-animated': animated,
```

Comment:
```suggestion
        'searchbar-animated': animated,
        'searchbar-disabled': this.disabled,
```

Reply:
I don't know why it's showing animated as a new line but that one shouldn't change :grin: